src/app/content-details/content-details.page.ts

```diff
@@ -1329,4 +1344,44 @@ showDeletePopup() {
       await loader.dismiss();
     }
   }
+
+  // pass coursecontext to ratinghandler if course is completed
+  async getContentState() {
+    return new Promise(async (resolve, reject) => {
+      this.courseContext = await this.preferences.getString(PreferenceKey.CONTENT_CONTEXT).toPromise();
```

Comment:
Why we need to check the course context here we already have a boolean isCourse? and ideally, this should be in a handler